msaad.go

```diff
@@ -470,31 +382,69 @@ func (m *MSAAD) SynchronizeUsers() error {
 	return nil
 }
 
+func (m *MSAAD) CreateOrUnarchiveUser(aadUser *msaadUser, aadEmailClean string) UserId {
```

Comment:
Will have to call bestemail again there, or move the logging down.

Reply:
Perhaps it is just easier to write out the full msaadUser...but it is an "Info" log...so will cause a bit of extra logging.

Reply:
Just added both mail and userprincipalname there. One reason for the logging is to be clear on the identifying details of the msaad user...since there were some misalignment and misidentification in the past.

Reply:
NOOOOOOOOOOOOOOO. You have a line `user := aadUser.profile.toAuthUser()`. The second line in `toAuthUser()` is literally a call to `bestEmail()` which is stored on `Email` on the `user`... please just use the `Email` on `user`...